test/framework/src/main/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -247,6 +251,15 @@ public void setFailReadsAfterUnblock(boolean failReadsAfterUnblock) {
         this.failReadsAfterUnblock = failReadsAfterUnblock;
     }
 
+    /**
+     * Enable blocking a single read of {@link org.elasticsearch.snapshots.SnapshotInfo} in case the repo is already blocked on another
+     * file. This allows testing very specific timing issues where a read of {@code SnapshotInfo} is much slower than another concurrent
+     * repository operation.
+     */
+    public void setBlockOnceOnReadSnapshotInfoIfAlreadyBlocked() {
```

Comment:
This is admittedly pretty :facepalm: but we don't have a clean way of expressing just a one-time block here. The dependency on another block is due to how the `blockExecution` loop works (doesn't play well with one-time actions).

Reply:
Could you add a reference to `MockRepository#blockExecution` in the javadoc? It took me a bit to understand how the dependency on another block was enforced.